subiquity/common/types.py

```diff
@@ -244,20 +244,29 @@ class Bootloader(enum.Enum):
 
 @attr.s(auto_attribs=True)
 class Partition:
-    size: int
-    number: int
-    annotations: List[str]
+    size: Optional[int] = None
+    number: Optional[int] = None
+    preserve: Optional[bool] = None
+    wipe: Optional[str] = None
+    annotations: Optional[List[str]] = []
+    mount: Optional[str] = None
+    format: Optional[str] = None
+    grub_device: Optional[bool] = None
```

Comment:
Is grub_device == None different from grub_device == False? Or would `grub_device: bool = False` be better here?

Reply:
It is different, actually.  Per J-P, it's easiest for the client if we pass around full Partition structure and not just the few elements that are relevant to an API call.  I'm using None as the 'just use the current value' sentinel for edit_partition, and a 'do what you think is best' indicator for add_partition.